ui/components/component-library/button-primary/button-primary.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import { ButtonBase } from '../button-base';
+import { BUTTON_SIZES } from './button-primary.constants';
+
+export const ButtonPrimary = ({
+  as = 'button',
+  block,
+  children,
+  className,
+  danger,
+  size = BUTTON_SIZES.MD,
+  icon,
+  iconPositionRight,
+  loading,
+  disabled,
+  iconProps,
+  ...props
+}) => {
+  return (
+    <ButtonBase
+      className={classnames(className, 'mm-button-primary', {
+        'mm-button-primary--type-danger': danger,
+      })}
+      {...props}
```

Comment:
They still need to be passed down to ButtonBase. Have used suggestion but will need formatting
```suggestion
      })}
block={block}
  danger={danger}
  size={size}
  icon={icon}
  iconPositionRight={iconPositionRight},
  loading={loading}
  disabled={disabled}
  iconProps={iconProps}
      {...props}
```